jetbrains-core/src/software/aws/toolkits/jetbrains/services/iam/Iam.kt

```diff
@@ -19,4 +23,19 @@ data class IamRole(val arn: String) {
     companion object {
         private val ARN_REGEX = "arn:.+:iam::.+:role/(.+)".toRegex()
     }
+}
+
+object IamResources {
+
+    private val LIST_RAW_ROLES = ClientBackedCachedResource(IamClient::class, "iam.list_roles") {
+        listRolesPaginator().roles().toList()
+    }
+
+    @JvmField
+    val LIST_LAMBDA_ROLES: Resource<List<IamRole>> = Resource.View(LIST_RAW_ROLES) {
+        filter { it.assumeRolePolicyDocument().contains(LAMBDA_PRINCIPAL) }
+            .map { IamRole(it.arn()) }
+            .sortedWith(Comparator.comparing<IamRole, String>(Function { t -> t.toString() }, String.CASE_INSENSITIVE_ORDER))
```

Comment:
The sort feels like an implementation detail that should not live at this layer

Reply:
the sorting? yeah we can remove that - the resource selector actually sorts by default

Reply:
removed